test.js

```diff
@@ -1,147 +1,107 @@
-// Simulate an API call to fetch tasks from a server
-function fetchTasks() {
-  return new Promise((resolve) => {
-    setTimeout(() => {
-      resolve([
-        { id: 1, text: 'Finish JavaScript project', completed: false },
-        { id: 2, text: 'Read a book', completed: true },
-        { id: 3, text: 'Go grocery shopping', completed: false },
-      ]);
-    }, 1000);
-  });
+// A non-optimized to-do list application with redundant logic and poor performance
+let toDoList = [];
```

Comment:
Ensure all element IDs are unique and correctly referenced to avoid potential DOM manipulation issues.